service/share/share.go

```diff
@@ -134,18 +135,24 @@ func (s *service) GetShares(context.Context, *Root) ([][]Share, error) {
 }
 
 func (s *service) GetSharesByNamespace(ctx context.Context, root *Root, nID namespace.ID) ([]Share, error) {
-	rowRootCIDs := make([]cid.Cid, 0)
+	rowRootCIDs := make(chan cid.Cid, len(root.RowsRoots))
+	wg := sync.WaitGroup{}
 	for _, row := range root.RowsRoots {
 		if !nID.Less(nmt.MinNamespace(row, nID.Size())) && nID.LessOrEqual(nmt.MaxNamespace(row, nID.Size())) {
-			rowRootCIDs = append(rowRootCIDs, plugin.MustCidFromNamespacedSha256(row))
+			wg.Add(1)
+			go func(row []byte) {
+				rowRootCIDs <- plugin.MustCidFromNamespacedSha256(row)
+			}(row)
```

Comment:
Pls look at `plugin.MustCidFromNamespacedSha256`. This is a cheap operation that does not need a parallelization. In fact, new code will be even slower and use more RAM due to goroutines overhead for an operation that can be done sequentially fast enough. Instead, you should run each `ipld.GetLeavesByNamaspace` in a separate routine, as it is a blocking operation that involves a lot of blocking IO.

Reply:
Thanks for explanations, @Wondertan . Sorry, for not getting what exactly should be done from the first attempt. May I use the same approach for `ipld.GetLeavesByNamaspace`(wg+channels) ?